plugins/GSdx/GSDevice.cpp

```diff
@@ -133,9 +133,9 @@ void GSDevice::Present(GSTexture* sTex, GSTexture* dTex, const GSVector4& dRect,
 
 GSTexture* GSDevice::FetchSurface(int type, int w, int h, bool msaa, int format)
 {
-	GSVector2i size(w, h);
+	const GSVector2i size(w, h);
 
-	for(list<GSTexture*>::iterator i = m_pool.begin(); i != m_pool.end(); ++i)
+	for(std::deque<GSTexture*>::const_iterator i = m_pool.cbegin(); i != m_pool.cend(); ++i)
```

Comment:
Ideally we can have only lookup for `size` if we keep the superset of `type`, `format` and `msaa` values as different pools, as current lookup code clearly says:
```C++
		if(t->GetType() == type && t->GetFormat() == format && t->GetSize() == size && t->IsMSAA() == msaa)
		// ...
```
Also we can rework `AgePool()` to remove all the old textures in a forward fashion,  so that the textures can be `push_back` instead of being `insert_front`: in this way we keep a (multidimensional) array of `vector` in which to store the textures to be pooled.
This would improve the speed of all texture pooling operations.